example/src/App.jsx

```diff
@@ -35,6 +35,10 @@ const App = () => {
     await new Promise(resolve => setTimeout(resolve, 1000));
   };
 
+  const handleDelete = async () => {
+    await new Promise(resolve => setTimeout(resolve, 1000));
+  };
```

Comment:
```suggestion
  const handleDelete = () => new Promise(resolve => setTimeout(resolve, 1000));
```
There is no need for additional async await wrap when you are already returning a promise.